File: framework/extracted/Ahorn/entities/SoulFragment.jl
module SoulFragment

using ..Ahorn, Maple

@mapdef Entity "MaggyHelper/SoulFragment" SoulFragment(x::Integer, y::Integer, fragmentId::String="", requiredSouls::Integer=3)

const placements = Ahorn.PlacementDict(
    "normal" => Ahorn.EntityPlacement(SoulFragment)
)

function Ahorn.selection(entity::SoulFragment)
    return Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::SoulFragment, room::Maple.Room)
    Ahorn.drawRectangle(ctx, Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16), (0.5, 0.5, 0.5, 0.5), (0.0, 0.0, 0.0, 1.0))
end

end

File: framework/extracted/Loenn/entities/soulFragment.lua
local utils = require("utils")
local drawableSprite = require("structs.drawable_sprite")

local soulFragment = {}

soulFragment.name = "MaggyHelper/SoulFragment"
soulFragment.depth = -100
soulFragment.fieldInformation = {
    requiredSouls = {
        minimumValue = 1
    }
}
soulFragment.placements = {
    {
        name = "normal",
        data = {
            fragmentId = "",
            requiredSouls = 3
        }
    }
}

function soulFragment.sprite(room, entity)
    local sprite = drawableSprite.fromTexture("objects/soul_fragment/hidden", entity)
    sprite:setPosition(entity.x, entity.y)
    sprite:setJustification(0.5, 0.5)
    sprite.color = {1.0, 1.0, 1.0, 0.9}
    return sprite
end

function soulFragment.selection(room, entity)
    return utils.rectangle(entity.x - 8, entity.y - 8, 16, 16)
end

return soulFragment
